lightning-background-processor/src/lib.rs

```diff
@@ -1144,6 +1186,304 @@ impl BackgroundProcessor {
+impl<
+		'a,
+		UL: 'static + Deref,
+		CF: 'static + Deref,
+		T: 'static + Deref,
+		F: 'static + Deref + Send,
+		G: 'static + Deref<Target = NetworkGraph<L>>,
+		L: 'static + Deref + Send,
+		P: 'static + Deref,
+		EH: 'static + EventHandler + Send,
+		PS: 'static + Deref + Send,
+		ES: 'static + Deref + Send,
+		M: 'static
+			+ Deref<
+				Target = ChainMonitor<<CM::Target as AChannelManager>::Signer, CF, T, F, L, P, ES>,
+			>
+			+ Send
+			+ Sync,
+		CM: 'static + Deref + Send,
+		OM: 'static + Deref + Send,
+		PGS: 'static + Deref<Target = P2PGossipSync<G, UL, L>> + Send,
+		RGS: 'static + Deref<Target = RapidGossipSync<G, L>> + Send,
+		PM: 'static + Deref + Send,
+		LM: 'static + Deref + Send,
+		S: 'static + Deref<Target = SC> + Send + Sync,
+		SC: for<'b> WriteableScore<'b>,
+		D: 'static + Deref,
+		O: 'static + Deref,
+		K: 'static + Deref,
+		OS: 'static + Deref<Target = OutputSweeperSync<T, D, F, CF, K, L, O>> + Send,
+	>
+	BackgroundProcessorConfigBuilder<
+		'a,
+		UL,
+		CF,
+		T,
+		F,
+		G,
+		L,
+		P,
+		EH,
+		PS,
+		ES,
+		M,
+		CM,
+		OM,
+		PGS,
+		RGS,
+		PM,
+		LM,
+		S,
+		SC,
+		D,
+		O,
+		K,
+		OS,
+	> where
+	UL::Target: 'static + UtxoLookup,
+	CF::Target: 'static + chain::Filter,
+	T::Target: 'static + BroadcasterInterface,
+	F::Target: 'static + FeeEstimator,
+	L::Target: 'static + Logger,
+	P::Target: 'static + Persist<<CM::Target as AChannelManager>::Signer>,
+	PS::Target: 'static + Persister<'a, CM, L, S>,
+	ES::Target: 'static + EntropySource,
+	CM::Target: AChannelManager,
+	OM::Target: AOnionMessenger,
+	PM::Target: APeerManager,
+	LM::Target: ALiquidityManager,
+	D::Target: ChangeDestinationSourceSync,
+	O::Target: 'static + OutputSpender,
+	K::Target: 'static + KVStore,
+{
+	/// Creates a new builder instance.
+	pub fn new(
+		persister: PS, event_handler: EH, chain_monitor: M, channel_manager: CM,
+		gossip_sync: GossipSync<PGS, RGS, G, UL, L>, peer_manager: PM, logger: L,
+	) -> Self {
+		Self {
+			persister,
+			event_handler,
+			chain_monitor,
+			channel_manager,
+			onion_messenger: None,
+			gossip_sync,
+			peer_manager,
+			liquidity_manager: None,
+			sweeper: None,
+			logger,
+			scorer: None,
+			_phantom: PhantomData,
+		}
+	}
+
+	/// Sets the optional onion messenger component.
+	pub fn with_onion_messenger(&mut self, onion_messenger: OM) -> &mut Self {
+		self.onion_messenger = Some(onion_messenger);
+		self
+	}
+
+	/// Sets the optional scorer component.
+	pub fn with_scorer(&mut self, scorer: S) -> &mut Self {
+		self.scorer = Some(scorer);
+		self
+	}
+
+	/// Sets the optional liquidity manager component.
+	pub fn with_liquidity_manager(&mut self, liquidity_manager: LM) -> &mut Self {
+		self.liquidity_manager = Some(liquidity_manager);
+		self
+	}
+
+	/// Sets the optional sweeper component.
+	pub fn with_sweeper(&mut self, sweeper: OS) -> &mut Self {
+		self.sweeper = Some(sweeper);
+		self
+	}
+
+	/// Builds and returns a [`BackgroundProcessorConfig`] object.
+	pub fn build(
```

Comment:
Maybe not doing the builder and just having a config object is worth it to cut down on all the generics and duplication?

Reply:
Ah, wrong on the copy part. Also no perf consequence obviously here.

But just leaving out `.build()` isn't that hard? It also isn't as if this config needs to be used in many places. Just once per project. To me all the duplication is already bad enough as it is, and getting rid of some by deviating a bit from a pattern is worth it to me.

Interested to hear @Anyitechs's opinion.